File: Mathematics/fibonacci/fibonacci_recursive.cpp
#include <bits/stdc++.h>
using namespace std;

int fib(int *arr, int n)
{
  if (n == 0 || n == 1)
    return 1;

  if (arr[n] > 0)
    return arr[n];

  int output = fib(arr, n - 1) + fib(arr, n - 2);
  arr[n] = output;
  return output;
}
int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  int n;
  cin >> n;
  int *input = new int[n];
  for (int i = 0; i < n; i++)
    input[i] = 0;

  cout << fib(input, n);
  delete[] input;
}

File: Mathematics/fibonacci/FibonacciRecursive.java
public class FibonacciRecursive {
    public static void main (String[]args) {
        System.out.println (nfib (0)); // Output - 0
        System.out.println (nfib (1)); // Output - 1
        System.out.println (nfib (2)); // Output - 1
        System.out.println (nfib (3)); // Output - 2
        System.out.println (nfib (4)); // Output - 3
        System.out.println (nfib (5)); // Output - 5
    }

    static int nfib (int n) {

        if (n == 0) return 0;
    
        if (n == 1) return 1;
    
        return nfib(n - 1) + nfib(n - 2);
    }
}
